src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs

```diff
@@ -15,10 +19,43 @@ namespace SFA.DAS.ProviderCommitments.Web.Controllers
     public class ApprenticeController : Controller
     {
         private readonly IModelMapper _modelMapper;
-     
-        public ApprenticeController(IModelMapper modelMapper)
+        private readonly ICurrentDateTime _currentDateTime;
+
+        public ApprenticeController(IModelMapper modelMapper,
+            ICurrentDateTime currentDateTime)
         {
             _modelMapper = modelMapper;
+            _currentDateTime = currentDateTime;
+        }
+
+        [Route("", Name = RouteNames.ManageApprentices)]
+        [DasAuthorize(ProviderFeature.ManageApprenticesV2)]
+        public async Task<IActionResult> Index(long providerId, ApprenticesFilterModel filterModel)
```

Comment:
We don't tend to have primitives as parameters to controller actions. You have a web request model below, GetApprenticeshipsRequest, which is then getting instantiated with these primitive values. This object ought to be in the method signature, not created in the action body. The setting of PageItemCount can be done in the mapper.

The name of the Request should match the action name, so this should be IndexRequest. To avoid collisions/confusion, we've started organising our models into subfolders, so this request could go into a Models\Apprentices\ folder. In fact, you can put the viewmodel in there too.

The model state validity check shouldn't be required.

Reply:
All moved as requested

Reply:
Removed ModelState check as well